src/raft/raft.rs

```diff
@@ -467,13 +483,18 @@ impl<T: Storage> Raft<T> {
 
     // bcast_heartbeat sends RPC, without entries to all the peers.
     pub fn bcast_heartbeat(&mut self) {
+        let ctx = self.read_only.last_pending_request_ctx();
+        self.bcast_heartbeat_with_ctx(ctx)
+    }
+
+    pub fn bcast_heartbeat_with_ctx(&mut self, ctx: Option<Vec<u8>>) {
```

Comment:
Then L955 should use `has_context` to check whether context exists. Using option here is redundant.